paddle/utils/PythonUtil.cpp

```diff
@@ -181,18 +185,29 @@ std::string getPyCallStack() {
   printPyErrorStack(os, true);
   return os.str();
 }
+
+PyObjectPtr import(const std::string &moduleName) {
+  auto module = PyImport_ImportModule(moduleName.c_str());
+  CHECK_PY(module) << "Import " << moduleName << "Error";
+  return PyObjectPtr(module);
+}
+
 }  // namespace py
 
 #endif
-
+extern "C" {
+extern const char enable_virtualenv_py[];
+}
 void initPython(int argc, char** argv) {
 #ifndef PADDLE_NO_PYTHON
   Py_SetProgramName(argv[0]);
   Py_Initialize();
   PySys_SetArgv(argc, argv);
-
   // python blocks SIGINT. Need to enable it.
   signal(SIGINT, SIG_DFL);
+
+  // Enable python virtualenv.
+  PyRun_SimpleString(enable_virtualenv_py);
```

Comment:
`virtualenv` is not need by us, but may be used by user.

These codes are used to `support running Paddle under a virtualenv`. And because Paddle process itself is a `python process`, so it should activate `virtualenv` when user have activated a `virtualenv` before.